src/app/core/header/header.component.html

```diff
@@ -1,12 +1,14 @@
 <header class="navbar navbar-dark navbar-fixed-top">
   <img src="assets/images/logo.png" alt="logo"/>
 
-  <ul class="nav justify-content-end">
+  <ul class="nav justify-content-end"
+      *ngIf="isAuthenticated()">
     <li class="nav-item">
-      <p class="nav-link">{{user.firstName}} {{user.lastName}}</p>
+      <p class="nav-link">{{getUserInfo()}}</p>
```

Comment:
It's not so good idea to put calculation into template. You will be faced performance issue, because change detection will be triggered so it cause of huge amount of recalculation.
It would be better to calculate this field at the initialization of component and then just get it and paint.